### ISSUE TITLE
Browser overrides do not work with relative paths

### ISSUE THREAD

**Issue by FredericHeem** [fuse-box/fuse-box#770]
When the library axios is being imported in a browser build, fuse-box also includes the files only used in a node build. Especially, the axios node build requires a `package.json`, which forces fuse-box to use the JSON plugin.

**Comment by nchanged** [comment]
@FredericHeem use target : "browser" if it includes `package.json` - nothing we can do about it

**Comment by FredericHeem** [comment]
FYI, the browser setting is set at https://github.com/FredericHeem/starhackit/blob/fusebox/client/fuse.js#L90
Same issue when `target: 'browser'` is adding to the `FuseBox` constructor.

**Comment by nchanged** [comment]
@FredericHeem you gotta set in the `FuseBox.init()` at least try that.
If it still requires `package.json` nothing you can do about it

**Comment by nchanged** [comment]
Speaking of your config
this [snippet](https://github.com/FredericHeem/starhackit/blob/fusebox/client/fuse.js#L39) is redundant 

http://fuse-box.org/page/quantum#process
```
console.log(process.env.VERSION)
```
will get replaced by Quantum
```
console.log("1.1.11")
```

And you don't need ENV_PLUGIN for that

`VERSION=1.1.11 node fuse`

**Comment by FredericHeem** [comment]
I've already tried to set `target: 'browser'` in `FuseBox ` without any success.
Regarding the version, I'd like to use the version inside `package.json`, not a version provided with an environment variable.

**Comment by nchanged** [comment]
Well then nothing we can do, it's in the source code

**Comment by FredericHeem** [comment]
`Webpack` handles a browser build of `axios` correctly so there should be a way to overcome this issue, shouldn't it?

**Comment by nchanged** [comment]
Listen, if there is a require statement to package.json - nothing we can, fusebox respects browser overrides.

If you insists show me the offending line, or prove that fuse doesn't respect it

**Comment by FredericHeem** [comment]
See https://github.com/mzabriskie/axios/issues/1068

**Comment by FredericHeem** [comment]
Please also have a look at https://github.com/mzabriskie/axios/blob/master/package.json#L71 
That's where the substitute `./lib/adapters/http.js` (node) with `./lib/adapters/xhr.js` (browser)

**Comment by nchanged** [comment]
Ok, I found the problem.

```
if (typeof XMLHttpRequest !== 'undefined') {
    // For browsers use XHR adapter
    adapter = require('./adapters/xhr');
  } else if (typeof process !== 'undefined') {
    // For node use HTTP adapter
    adapter = require('./adapters/http');
  }
```

Altho it's replaced for the browser, FuseBox still bundles it, because `typeof process` if defined.
So in a way, it will be bundled anyway because of that condition. I thought that the browser field fixes it, but it turns out the statement is used elsewhere.

**Comment by nchanged** [comment]
But I think your point is right, fuse is not resolving a path before comparing it to the overrides. Good catch

**Comment by nchanged** [comment]
Thank you for finding this, I made a very important change, now it's working properly in `fuse-box@2.2.3-beta.21` Please test

**Comment by FredericHeem** [comment]
Great success, thanks a lot.

**Comment by xaviergonz** [comment]
The fix broke this one on 2.2.3: https://github.com/fuse-box/fuse-box/issues/711

**Comment by nchanged** [comment]
#776
